dlpy/network.py

```diff
@@ -726,22 +745,68 @@ def load_weights_from_file(self, path, format_type='KERAS'):
             contains the weight table.
         format_type : KERAS, CAFFE
             Specifies the source framework for the weights file
+        data_spec: list of :class:`DataSpec`, optional
+            data specification for input and output layer(s)            
 
         '''
         cas_lib_name, file_name = caslibify(self.conn, path, task='load')
 
-        self._retrieve_('deeplearn.dlimportmodelweights', model=self.model_table,
-                        modelWeights=dict(replace=True,
-                                          name=self.model_name + '_weights'),
-                        formatType=format_type, weightFilePath=file_name,
-                        caslib=cas_lib_name)
+        if (data_spec):
+            from io import StringIO
+            import sys
+
+            # redirect stderr
+            old_stderr = sys.stderr
+            sys.stderr = stderr = StringIO()
+
+            # run action with dataSpec option
+            rt = self._retrieve_('deeplearn.dlimportmodelweights', 
+                                message_level='NONE',
+                                model=self.model_table,
+                                modelWeights=dict(replace=True, name=self.model_name + '_weights'),
+                                dataSpecs=data_spec,
+                                formatType=format_type, weightFilePath=file_name, caslib=cas_lib_name,
+                                #_debug = dict(nodes="all",ranks='-1',roles="all",display="d7a752.na.sas.com:0")
+                                );
+            
+            # restore stderr
+            sys.stderr = old_stderr                
```

Comment:
Same as above.

Reply:
Please see my reply to your first comment on this topic.